core/backend/src/annotations/FrameGeometry.ts

```diff
@@ -0,0 +1,324 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module ElementGeometry
+ */
+
+
+import { Id64 } from "@itwin/core-bentley";
+import { BackgroundFill, ColorDef, ElementGeometry, FillDisplay, GeometryParams, TextAnnotationFrameShape, TextFrameStyleProps } from "@itwin/core-common";
+import { Angle, AngleSweep, Arc3d, LineString3d, Loop, Path, Point3d, Range2d, Transform, Vector2d } from "@itwin/core-geometry";
+
+/**
+ * This namespace contains methods for computing the geometry a frame that encloses the range of a text annotation.
+ * @beta
+ */
+export namespace FrameGeometry {
+
+  /**
+   * Based on the frame style, this method will construct and append [[GeometryParams]] (for line style) and a [[Loop]] (for the frame shape) to the builder.
+   * @param builder that will be appended to in place
+   * @param frame
+   * @param range to enclose with the frame
+   * @param transform that transform the range to world coordinates
+   * @returns `true` if any geometry was appended to the builder
+  * @beta
+   */
+  export const appendFrameToBuilder = (builder: ElementGeometry.Builder, frame: TextFrameStyleProps, range: Range2d, transform: Transform, geomParams?: GeometryParams): boolean => {
+    if (frame.shape === "none") {
+      return false;
+    }
+
+    const params = geomParams?.clone() ?? new GeometryParams(Id64.invalid);
```

Comment:
Why `Id64.invalid`? Does anyone actually pass in `undefined` for `geomParams`?

Reply:
I can't remember where I got that from. Should I be using the element's category id?

Reply:
Okay I remember where I got that from. The ElementGeometry.builder and GeometryStreamBuilder both use Id64.Invalid for the lines in the fractions bars.

Maybe the frame geom should be a different category though. Thoughts @pmconne?

Reply:
That's for your app and/or the user to decide. Either way, I think your API here should require the caller to configure all of that - you certainly can't decide here what category to put the frame on. If you want to continue just cloning the input GeometryParams, I think you should make them non-optional so you can eliminate Id64.invalid here.

Reply:
Done! 

I think that should be the last comment to address!